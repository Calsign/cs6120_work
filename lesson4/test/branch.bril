@main() {
  x: int = const 4;
  cond: bool = const false;
  br cond .br1 .br2;
.br1:
  y: int = const 2;
  z: int = const 4;
  jmp .end;
.br2:
  y: int = const 2;
  x: int = const 3;
  jmp .end;
.end:
  print x;
}
