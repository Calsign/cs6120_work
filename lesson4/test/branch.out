{"op":"const","value":4,"dest":"x","type":"int"}  ->  x: 4
{"op":"const","value":false,"dest":"cond","type":"bool"}  ->  cond: false, x: 4
{"op":"br","args":["cond"],"labels":["br1","br2"]}  ->  cond: false, x: 4
{"op":"const","value":2,"dest":"y","type":"int"}  ->  cond: false, x: 4, y: 2
{"op":"const","value":4,"dest":"z","type":"int"}  ->  cond: false, x: 4, y: 2, z: 4
{"op":"jmp","labels":["end"]}  ->  cond: false, x: 4, y: 2, z: 4
{"op":"const","value":2,"dest":"y","type":"int"}  ->  cond: false, x: 4, y: 2
{"op":"const","value":3,"dest":"x","type":"int"}  ->  cond: false, x: 3, y: 2
{"op":"jmp","labels":["end"]}  ->  cond: false, x: 3, y: 2
{"op":"print","args":["x"]}  ->  cond: false, x: ?, y: 2, z: 4
